spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -320,27 +369,60 @@ def main(argv=None):
                                          angle_correction=angle_correction,
                                          param_centerline=param_centerline,
                                          verbose=verbose)
+    if fname_pmj is not None:
+        im_ctl, mask, slices, centerline = get_slices_for_pmj_distance(fname_segmentation, fname_pmj,
+                                                                       distance_pmj, extent_mask,
+                                                                       param_centerline=param_centerline,
+                                                                       verbose=verbose)
+        # Save mask
+        fname_mask_out = add_suffix(arguments.i, '_mask_csa')
+        mask.save(fname_mask_out)
+        # Save extrapolated centerline
+        fname_ctl = add_suffix(arguments.i, '_centerline_extrapolated')
+        im_ctl.save(fname_ctl)
+        # Save array of the centerline in a .csv file
+        fname_ctl_csv, _ = splitext(add_suffix(arguments.i, '_centerline_extrapolated'))
+        np.savetxt(fname_ctl_csv + '.csv', centerline, delimiter=",")
+        # Generated centerline smoothed in RL direction for visualization (and QC report)
+        fname_ctl_smooth = add_suffix(fname_ctl, '_smooth')
+        sct_maths.main(['-i', fname_ctl, '-smooth', '10,1,1', '-o', fname_ctl_smooth])
```

Comment:
All of these steps are just for visualization/feedback, right? 

Do they _need_ to be saved to files? I ask because it seems kind of noisy to pollute the user's working directory, especially since the `sct_process_segmentation` is mainly just meant for metric calculation.

So, I'm wondering if maybe the QC report good enough by itself for this kind of feedback.

Reply:
The mask and extrapolated centerline are only relevant (for now) for th QC report. Maybe I will use them for a follow up project on this to validate the method, but I don't know yet, haven't planed that much of it!
I use the CSV file of the extrapolated centerline for my research after (e.g. compute distance between C2-C3 disc and PMJ)
we could hide them all in the behind `-verbose 2`, 

this way, we would save the centerline and the mask in a tmp directory?

Reply:
> this way, we would save the centerline and the mask in a tmp directory?

I think that sound reasonable! :smile: And maybe move the file-saving over to the part of the PR where QC is generated, just to make it clearer that they only need to be generated if QC is requested.

Reply:
done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3478/commits/e17592974a7def4671cd353e666ac91a5c8fb1cc